src/apps/replication/lib/replica.cpp

```diff
@@ -123,8 +125,9 @@ void replica::on_client_read(const read_request_header& meta, dsn_message_t requ
 
     if (meta.semantic == read_semantic_t::ReadLastUpdate)
     {
-        if (status() != PS_PRIMARY || 
-            last_committed_decree() < _primary_states.last_prepare_decree_on_new_primary)
+        if (status() != PS_PRIMARY /*||
+            last_committed_decree() < _primary_states.last_prepare_decree_on_new_primary */)
```

Comment:
there is a short period where a new primary may not have the Last-Updated local state , in this case we need to wait until the state is catched up